main.py

```diff
@@ -95,17 +180,43 @@ def draw() -> Shape:
         if events.type == generate_new_round and current_round <= 24:
             current_round = current_round + 1
 
-            if current_round > 24:
+            if current_round > 44:
```

Comment:
You may need to explain the round logic again, I can see we have a maximum of 81 rounds, but we're ending the game and saving the results at round 45 or higher.

Reply:
Probably a mistake, need to look over this again to understand my own logic